models/gif_psc_exp_multisynapse.h

```diff
@@ -0,0 +1,422 @@
+
+  References:
+
+  [1] Mensi S, Naud R, Pozzorini C, Avermann M, Petersen CC, Gerstner W (2012)
+  Parameter extraction and classification of three cortical neuron types
+  reveals two distinct adaptation mechanisms. J. Neurophysiol., 107(6),
+  1756-1775.
+
+  [2] Pozzorini C, Mensi S, Hagens O, Naud R, Koch C, Gerstner W (2015)
+  Automated High-Throughput Characterization of Single Neurons by Means of
+  Simplified Spiking Models. PLoS Comput. Biol., 11(6), e1004275.
+
+
+  Sends: SpikeEvent
+
+  Receives: SpikeEvent, CurrentEvent, DataLoggingRequest
+
+  Author: March 2016, Setareh
+  SeeAlso: pp_psc_delta, gif_psc_exp, gif_cond_exp, gif_cond_exp_multisynapse */
+
+namespace nest
+{
+
+class gif_psc_exp_multisynapse : public Archiving_Node
+{
+
+public:
+  gif_psc_exp_multisynapse();
+  gif_psc_exp_multisynapse( const gif_psc_exp_multisynapse& );
+
+  /**
+   * Import sets of overloaded virtual functions.
+   * @see Technical Issues / Virtual Functions: Overriding, Overloading, and
+   * Hiding
+   */
+  using Node::handle;
+  using Node::handles_test_event;
+
+  port send_test_event( Node&, rport, synindex, bool );
+
+  void handle( SpikeEvent& );
+  void handle( CurrentEvent& );
+  void handle( DataLoggingRequest& );
+
+  port handles_test_event( SpikeEvent&, rport );
+  port handles_test_event( CurrentEvent&, rport );
+  port handles_test_event( DataLoggingRequest&, rport );
+
+  void get_status( DictionaryDatum& ) const;
+  void set_status( const DictionaryDatum& );
+
+private:
+  void init_state_( const Node& proto );
+  void init_buffers_();
+  void calibrate();
+
+  void update( Time const&, const long_t, const long_t );
+
+  // The next two classes need to be friends to access the State_ class/member
+  friend class RecordablesMap< gif_psc_exp_multisynapse >;
+  friend class UniversalDataLogger< gif_psc_exp_multisynapse >;
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Independent parameters of the model.
+   */
+  struct Parameters_
+  {
+    double_t g_L_;
+    double_t E_L_;
+    double_t V_reset_;
+    double_t Delta_V_;
+    double_t V_T_star_;
+    double_t lambda_0_; /** 1/ms */
+
+
+    /** Refractory period in ms. */
+    double_t t_ref_;
+
+    /** Membrane capacitance in pF. */
+    double_t c_m_;
+
+    /** We use stc and sfa, respectively instead of eta and gamma
+    (mentioned in the references). */
+
+    /** List of spike triggered current time constant in ms. */
+    std::vector< double_t > tau_stc_;
+
+    /** List of spike triggered current jumps in nA. */
+    std::vector< double_t > q_stc_;
+
+    /** List of adaptive threshold time constant in ms. */
+    std::vector< double_t > tau_sfa_;
+
+    /** List of adaptive threshold jumps in mV. */
+    std::vector< double_t > q_sfa_;
+
+    /** Time constants of synaptic currents in ms */
+    std::vector< double_t > tau_syn_;
+
+    /** type is long because other types are not put through in GetStatus */
```

Comment:
Doxygen comments beginning with `/**` should describe the member, not explain design decisions. Use normal comments for that.